packages/deployer/internal/process-contracts.js

```diff
@@ -1,60 +1,61 @@
 const fs = require('fs/promises');
-const { getContractAST } = require('@synthetixio/core-js/utils/hardhat/artifacts');
+const { parseFullyQualifiedName } = require('hardhat/utils/contract-names');
 const { getBytecodeHash } = require('@synthetixio/core-js/utils/ethers/contracts');
 const { findInheritedContractNames } = require('@synthetixio/core-js/utils/ast/finders');
-const { contractIsModule, getContractFilePath } = require('./contract-helper');
+const { getContractFilePath } = require('./contract-helper');
 
 /**
  * Initialize contract metadata on hre.deployer.deployment.*
  * This will in turn save all the necessary data to deployments file.
- * @param {string} contractName
+ * @param {string} contractFullyQualifiedName
  */
-async function initContractData(contractName) {
+async function initContractData(contractFullyQualifiedName, extraData = {}) {
```

Comment:
Added the option to add custom data when initializing it on deplouyment file. This gives us the option of adding custom markers like `isProxy` & `isRouter`, which I used for some refactors throughout the PR.